charts/vela-core/templates/defwithtemplate/cpuscaler.yaml

```diff
@@ -14,38 +14,30 @@ spec:
     cue:
       template: |
         outputs: cpuscaler: {
-        	apiVersion: "autoscaling/v2beta2"
+        	apiVersion: "autoscaling/v1"
         	kind:       "HorizontalPodAutoscaler"
         	metadata: name: context.name
         	spec: {
         		scaleTargetRef: {
-        			apiVersion: "apps/v1"
-        			kind:       "Deployment"
+        			apiVersion: parameter.targetAPIVersion
+        			kind:       parameter.targetKind
         			name:       context.name
         		}
-        		minReplicas: parameter.min
-        		maxReplicas: parameter.max
-        		metrics: [{
-        			type: "Resource"
-        			resource: {
-        				name: "cpu"
-        				target: {
-        					type:               "Utilization"
-        					averageUtilization: parameter.cpuUtil
-        				}
-        			}
-        		}]
+        		minReplicas:                    parameter.min
+        		maxReplicas:                    parameter.max
+        		targetCPUUtilizationPercentage: parameter.cpuUtil
         	}
         }
         parameter: {
-
         	// +usage=Specify the minimal number of replicas to which the autoscaler can scale down
         	min: *1 | int
-
         	// +usage=Specify the maximum number of of replicas to which the autoscaler can scale up
         	max: *10 | int
-
-        	// +usage=Specify the average cpu utilization, for example, 50 means the CPU usage is 50%
+        	// +usage=Specify the average CPU utilization, for example, 50 means the CPU usage is 50%
         	cpuUtil: *50 | int
+        	// +usage=Specify the apiVersion of scale target
+        	targetAPIVersion: *"apps/v1" | string
+        	// +usage=Specify the kind of scale target
+        	targetKind: *"Deployment" | string
```

Comment:
Should field `spec.appliesToWorkloads` be set for this trait?

Reply:
The `appliesToWorkloads` is actually a useless field right now and won't affect the resources(only checked in webhook).

Reply:
It's a field of `vela traits`. Leaving it as unset means it can apply to any workload. But it was not true.

Reply:
I add `targetAPIVersion` and `targetKind` as optional fields, it user specify the values, it can apply to any workload.

Reply:
In which case should users define targetAPIVersion and Kind?

Reply:
If the users uses this trait to non-deployment resources.